### ISSUE TITLE
Slider DPAD support for Android TV

### ISSUE THREAD

**Issue by andraskende** [flutter/flutter#54984]
I am trying to use slider in a video player on Android TV app, testing on Nvidia Shield TV
just upgraded master, with the #53945 change i can now use the slider with remote d-pad, but there  are some issues with DPAD on Android TV platform with remote dpad arrow keys:

- slider once focused can't unfocus / move to other widget buttons on screen...
- DPAD left and bottom rewinds, right and top forwards but DPAD could be left: rewind, right: forward, top and bottom moves to other widgets...

Also I think another enhancement would be great to add onPressed() to Slider.
use case is a video player where the slider button press pauses / plays video (this is how youtube android tv app functions..) 

Thanks !!

related:
https://github.com/flutter/flutter/pull/53945

**Comment by Mi2Sir** [comment]
I meet the same problem.
Any solution now?

**Comment by gspencergoog** [comment]
@JoseAlba or @clocksmith can you take a look at this?

**Comment by zanderso** [comment]
[Triage] @gspencergoog What's the status of this issue? What is left to do?

**Comment by YaarPatandarAA** [comment]
Any Updates on how to solve the issues stated in the initial post? Any Workarounds?

**Comment by gspencergoog** [comment]
As far as I know, the DPAD key events should be interceptable on Android, so that they will work even when a text field is focused. Adding the `onPressed` to the slider I don't think has been done (or evaluated).

**Comment by Prince-of-death** [comment]
> I am trying to use slider in a video player on Android TV app, testing on Nvidia Shield TV just upgraded master, with the #53945 change i can now use the slider with remote d-pad, but there are some issues with DPAD on Android TV platform with remote dpad arrow keys:
> 
> * slider once focused can't unfocus / move to other widget buttons on screen...
> * DPAD left and bottom rewinds, right and top forwards but DPAD could be left: rewind, right: forward, top and bottom moves to other widgets...
> 
> Also I think another enhancement would be great to add onPressed() to Slider. use case is a video player where the slider button press pauses / plays video (this is how youtube android tv app functions..)
> 
> Thanks !!
> 
> related: #53945

wrap th slider like this
```dart
MediaQuery(
      data: MediaQueryData(
        navigationMode: NavigationMode.directional,
      ),
      child: Slider(),
),
```

**Comment by Blacktaler** [comment]
> > I am trying to use slider in a video player on Android TV app, testing on Nvidia Shield TV just upgraded master, with the #53945 change i can now use the slider with remote d-pad, but there are some issues with DPAD on Android TV platform with remote dpad arrow keys:
> > 
> > * slider once focused can't unfocus / move to other widget buttons on screen...
> > * DPAD left and bottom rewinds, right and top forwards but DPAD could be left: rewind, right: forward, top and bottom moves to other widgets...
> > 
> > Also I think another enhancement would be great to add onPressed() to Slider. use case is a video player where the slider button press pauses / plays video (this is how youtube android tv app functions..)
> > Thanks !!
> > related: #53945
> 
> wrap th slider like this
> 
> ```dart
> MediaQuery(
>       data: MediaQueryData(
>         navigationMode: NavigationMode.directional,
>       ),
>       child: Slider(),
> ),
> ```

Very helpful, thank you. I was experiencing this problem and just reviewed Slider. I saw NavigationMode in slider file, but I didn't know how to control or change its value. Thank you, this is working solution.
